lib/src/components/NxList/__tests__/NxListButtonItem.test.tsx

```diff
@@ -5,76 +5,119 @@
   });
 
-  it('renders the classNames given to both the li and the button', function() {
-    const contentEl = getShallow({className: 'customClassLi', buttonClassName: 'customClassBtn'});
+  it('sets the specified classnames to the list in addition to the defaults', function() {
+    const el = renderEl({ className: 'foo' }),
+        defaultEl = renderEl()!;
 
-    expect(contentEl).toExist();
-    expect(contentEl).toMatchSelector('.nx-list__item.nx-list__item--clickable.customClassLi');
-    expect(contentEl.find('button')).toMatchSelector('.nx-list__btn.customClassBtn');
+    expect(el).toHaveClass('foo');
+
+    for (const cls of Array.from(defaultEl.classList)) {
+      expect(el).toHaveClass(cls);
+    }
   });
 
-  it('renders a clickable list whose list item element has prop disabled and is disabled', function() {
-    const contentEl = getMounted({disabled: true});
+  it('sets the specified attrs to the list', function() {
+    const el = renderEl({ id: 'foo', lang: 'en' });
 
-    expect(contentEl).toExist();
-    expect(contentEl).toHaveProp('disabled');
-    expect(contentEl).toBeDisabled();
+    expect(el).toHaveAttribute('id', 'foo');
+    expect(el).toHaveAttribute('lang', 'en');
   });
 
-  it('renders a clickable list whose list item element has prop selected and is selected', function() {
-    const contentEl = getMounted({selected: true});
+  it('sets the specified classnames to the button in addition to the defaults ' +
+    'when buttonClassName prop is provided', function() {
+    const btn = quickRender({ buttonClassName: 'boo' }).getByRole('button'),
+        defaultBtn = quickRender().getByRole('button');
+
+    expect(btn).toHaveClass('boo');
 
-    expect(contentEl).toExist();
-    expect(contentEl).toHaveProp('selected');
-    expect(contentEl.find('button')).toHaveClassName('selected');
+    for (const cls of Array.from(defaultBtn.classList)) {
+      expect(btn).toHaveClass(cls);
+    }
   });
 
-  it('renders children correctly', function() {
-    const children = [
-      <NxList.Text key="1">Test Item 1 Text</NxList.Text>,
-      <NxList.Subtext key="2">Test Item 1 Subtext</NxList.Subtext>
-    ];
-    const contentEl = getMounted({children});
+  it('sets the specified attrs to the button ' +
+    'when buttonAttributes prop is provided', function() {
+    const buttonAttributes = { id: 'boo', value: 'test' },
+        btn = quickRender({ buttonAttributes }).getByRole('button');
 
-    expect(contentEl).toExist();
-    expect(contentEl.find('span').at(0)).toMatchSelector('span.nx-list__text');
-    expect(contentEl.find('span').at(0)).toHaveText('Test Item 1 Text');
-    expect(contentEl.find('span').at(1)).toMatchSelector('span.nx-list__subtext');
-    expect(contentEl.find('span').at(1)).toHaveText('Test Item 1 Subtext');
+    expect(btn).toHaveAttribute('id', 'boo');
+    expect(btn).toHaveAttribute('value', 'test');
+  });
 
+  it('sets disabled and aria-disabled="true" to the button ' +
```

Comment:
I don't think `aria-disabled` actually matters in this case; I'd only check `.toBeDisabled()` here.  We do however need a test that `aria-disabled` gets set if the caller passes in `disabled` as one of the classnames

Reply:
Ah right, make sense. Updated here eb1b5df24c158cfcbef7fc0843ac76e04ae9717e